lib/utils.bash

```diff
@@ -36,26 +38,27 @@ list_all_versions() {
 }
 
 download_release() {
-	local version filename url
-	version="$1"
-	filename="$2"
+	local install_type="$1"
+	local version="$2"
+	local filename="$3"
+	local url
 
-	# TODO: Adapt the release URL convention for <YOUR TOOL>
-	url="$GITHUB_REPO/archive/refs/tags/v${version}.tar.gz"
+	if [ "$install_type" == "version" ]; then
+		url="${GITHUB_REPO}/archive/refs/tags/v${version}.tar.gz"
+	else
+		url="https://api.github.com/repos/${OWNER}/${REPO}/tarball/${version}"
```

Comment:
```suggestion
		url="https://${GITHUB_API_URL}/repos/${OWNER}/${REPO}/tarball/${version}"
```

Reply:
also, should the filename have the extension? `/tarball/${version}.tar.gz`

Reply:
filename is generated in a different file `/bin/download`
`release_file="$ASDF_DOWNLOAD_PATH/$RELEASE_NAME-$ASDF_INSTALL_VERSION.tar.gz"`

The filename isn't too important I guess since it just gets cleaned up after.